src/websocket/websocket.c

```diff
@@ -364,14 +364,14 @@ web_socket_util_parse_headers (const gchar *data,
           g_strstrip (name);
           value = g_strndup (colon + 1, line - (colon + 1));
           g_strstrip (value);
-          g_hash_table_insert (parsed_headers, name, value);
 
           if (!is_valid_line (name, -1) || !g_utf8_validate (value, -1, NULL))
             {
               g_debug ("received invalid header");
               consumed = -1;
               break;
```

Comment:
Exiting the loop without inserting the freshly-allocated `name` and `value` into the table is a leak.

A better approach is probably to move the declarations of the `name` and `value`  inside the loop, use `g_autofree` on them, and use `g_steal_pointer()` at the point of inserting them into the hash table.

Reply:
Specifically: move the variable declaration to the site of the assignment.

Reply:
I hope that I've done this correctly. Sorry it's my first time using g_autofree and g_steal_pointer